### PR TITLE
RPP: Clean up legacy Fraud_Prevention_Service

### PR DESCRIPTION
**Pull request by htdat**
Fixes #

## Notes

- A part of #7413 
- Related PR #7696 
- ⚠️ ⚠️ ⚠️  Will merge to `develop` only in Jan 2024

#### Testing instructions

*

-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7697` or branch name `rpp/clean-up-legacy-fraud-service-base-7696` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=rpp/clean-up-legacy-fraud-service-base-7696) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9e0e21c67255d8ca76a9c564a7c00bed1e2da702
- Build time: 2023-11-17 09:51:50 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-wc-payment-gateway-wcpay.php [hunk 1]
```diff
@@ -913,16 +914,14 @@ public function process_payment( $order_id ) {
...
-					);
-				}
+
+			$fraud_prevention_service = wcpay_get_container()->get( PaymentFraudPreventionService::class );
+			// phpcs:ignore WordPress.Security.NonceVerification.Missing,WordPress.Security.ValidatedSanitizedInput.MissingUnslash,WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			if ( $fraud_prevention_service->is_enabled() && ! $fraud_prevention_service->verify_token( $_POST['wcpay-fraud-prevention-token'] ?? null ) ) {
+				throw new Process_Payment_Exception(
+					__( "We're not able to process this payment. Please refresh the page and try again.", 'woocommerce-payments' ),
+					'fraud_prevention_enabled'
+				);
```
Comment (htdat) [inline includes/class-wc-payment-gateway-wcpay.php]: ## Check the logic of removing `WC()->session` here and other places in this PR

Main concern: https://github.com/Automattic/woocommerce-payments/pull/5393

It looks fine as the new `src` class `FraudPreventionService` uses `SessiionService`, which handles nicely when `WC()->session` is not yet initialized. But it's still worth double-checking.

**Comment by naman03malhotra** [comment]
I see it is blocked, maybe we can remove it from this sprint?

**Comment by htdat** [comment]
> I see it is blocked, maybe we can remove it from this sprint?

Yes, removed it. But I also add this to Epic https://github.com/Automattic/woocommerce-payments/issues/6673 in case we revist our work for project "Payment Process Re-Engineering"

**Comment by htdat** [comment]
Closing as we're not going to reopen project "Payment Process Re-Engineering"

**Pull request closed without merging**